FastSurferCNN/run_prediction.py

```diff
@@ -0,0 +1,419 @@
+
+# Copyright 2019 Image Analysis Lab, German Center for Neurodegenerative Diseases (DZNE), Bonn
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# IMPORTS
+import numpy as np
+import torch
+import os
+import logging
+import nibabel as nib
+import time
+import glob
+import sys
+print("Run pred", sys.path)
+
+from eval import Inference
+from utils.load_config import load_config
+import sys
+
+import data_loader.data_utils as du
+import data_loader.conform as conf
+
+##
+# Global Variables
+##
+LOGGER = logging.getLogger("eval")
+LOGGER.setLevel(logging.DEBUG)
+LOGGER.addHandler(logging.StreamHandler(stream=sys.stdout))
+
+
+##
+# Processing
+##
+def set_up_cfgs(cfg, args):
+    cfg = load_config(cfg)
+    cfg.OUT_LOG_DIR = args.out_dir if args.out_dir is not None else cfg.LOG_DIR
+    cfg.OUT_LOG_NAME = "fastsurfer"
+    cfg.TEST.BATCH_SIZE = args.batch_size
+
+    cfg.MODEL.OUT_TENSOR_WIDTH = cfg.DATA.PADDED_SIZE
+    cfg.MODEL.OUT_TENSOR_HEIGHT = cfg.DATA.PADDED_SIZE
+    return cfg
+
+
+##
+# Input array preparation
+##
+class RunModelOnData:
+
+    def __init__(self, args):
+        self.current_subject = ""
+        self.subject_name = ""
+        self.gt, self.gt_data, self.orig, self.orig_data = "", "", "", ""
+        self.sf = 1.0
+        self.s = ""
+        self.out_dir = args.out_dir
+        self.orig_filename = os.path.join(self.current_subject, args.orig_name)
+
+        if args.gt_dir is None:
+            self.gt_filename = os.path.join(self.current_subject, args.gt_name)
+        else:
+            self.gt_filename = os.path.join(args.gt_dir, self.subject_name, args.gt_name)
+
+        if args.out_dir is not None:
+            self.pred_name = os.path.join(args.out_dir, self.subject_name, args.pred_name)
+        else:
+            self.pred_name = os.path.join(self.current_subject, args.pred_name)
+
+        self.conf_name = args.conf_name
+
+        if args.run_viewagg_on == "gpu":
+            # run view agg on the gpu (force)
+            self.small_gpu = False
+
+        elif args.run_viewagg_on == "check":
+            # check, if GPU is big enough to run view agg on it
+            # (this currently takes only the total memory into account, not the occupied on)
+            total_gpu_memory = sum([torch.cuda.get_device_properties(i).__getattribute__("total_memory") for i in
+                                    range(torch.cuda.device_count())])
+            self.small_gpu = total_gpu_memory < 8000000000
```

Comment:
this number needs to be adjusted for VINN, it now depends on the size of the image:
formula for memory required for view aggregation:
`width (up to 320) * height (up to 320) * depth (up to 320) * classes (79) * sizeof(float) (4)` which is about 9.64 GB